packages/actor-abstract-path/lib/ActorAbstractPath.ts

```diff
@@ -34,17 +35,30 @@ export abstract class ActorAbstractPath extends ActorQueryOperationTypedMediated
   }
 
   // Generates a blank node that does not yet occur in the path
-  public generateBlankNode(path?: Algebra.Path, name?: string): BlankNode {
+  public generateVariable(path?: Algebra.Path, name?: string): Variable {
     if (!name) {
-      return this.generateBlankNode(path, 'b');
+      return this.generateVariable(path, 'b');
     }
 
     // Path predicates can't contain variables/blank nodes
     if (path && (path.subject.value === name || path.object.value === name)) {
-      return this.generateBlankNode(path, `${name}b`);
+      return this.generateVariable(path, `${name}b`);
     }
 
-    return blankNode(name);
+    return variable(name);
+  }
+
+  // Such connectivity matching does not introduce duplicates (it does not incorporate any count of the number
+  // of ways the connection can be made) even if the repeated path itself would otherwise result in duplicates.
+  // https://www.w3.org/TR/sparql11-query/#propertypaths
+  public isPathArbitraryLengthDistinct(context: ActionContext): ActionContext {
+    if (!context || !context.get('isPathArbitraryLengthDistinct')) {
```

Comment:
Could you put `'isPathArbitraryLengthDistinct'` in a static variable on this class, and always refer to that one? This will make it impossible to have typo's.